### DISCUSSION TITLE
How to check a value used in in the run?

### DISCUSSION THREAD

**Discussion by georgemilosh** [georgemilosh/Climate-Learning#51] [Q&A] [answered]
So when the run has been initiated by `Learn2_new.py` we have two relevant files `config.json` and `runs.json`, where `runs.json` overwrite some arguments in `config.json`. Do you have a ready-made function which automatically extracts the correct value based on both of these files?

**Comment by georgemilosh** [comment]
@AlessandroLovo , do you have a quick suggestion? You probably faced with this before

**Comment by AlessandroLovo** [comment, marked as answer]
There is no ready-made function, but the implementation is relatively easy.

```
folder = 'myfolder'
config_dict = ut.json2dict(f'{folder}/config.json')
runs = ut.json2dict(f'{folder}/runs.json')
```
Let's say now you are interested in the first run
```
run = list(runs.values())[0]
```
Then to extract the argument `arg`, you can do as follows:
```
run_kwargs = ut.set_values_recursive(config_dict, run['args'])
arg_value = ut.extract_nested(run_kwargs, arg)
```

A one liner option could instead be
```
arg_value = ut.extact_nested(run['args'], arg, ut.extract_nested(config_dict, arg))
```
The third argument of `ut.extract_nested` is the default value to return if it is not present in the dicionary
